chainercv/links/model/pspnet/pspnet.py

```diff
@@ -0,0 +1,476 @@
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet101_VOC2012_473_reference.npz'
+        },
+        'cityscapes': {
+            'n_class': 19,
+            'input_size': (713, 713),
+            'n_blocks': [3, 4, 23, 3],
+            'feat_size': 90,
+            'mid_stride': True,
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet101_cityscapes_713_reference.npz'
+        },
+        'ade20k': {
+            'n_class': 150,
+            'input_size': (473, 473),
+            'n_blocks': [3, 4, 6, 3],
+            'feat_size': 60,
+            'mid_stride': True,
+            'pyramids': [6, 3, 2, 1],
+            'mean': np.array([123.68, 116.779, 103.939]),
+            'url': 'https://github.com/mitmul/chainer-pspnet/releases/download'
+                   '/ChainerCV_PSPNet/pspnet50_ADE20K_473_reference.npz'
+        }
+    }
+
+    def __init__(self, n_class=None, input_size=None, n_blocks=None,
+                 pyramids=None, mid_stride=None, mean=None, comm=None,
+                 pretrained_model=None, initialW=None):
+        super(PSPNet, self).__init__()
+
+        if pretrained_model in self._models:
+            if 'n_class' in self._models[pretrained_model]:
+                n_class = self._models[pretrained_model]['n_class']
+            if 'input_size' in self._models[pretrained_model]:
+                input_size = self._models[pretrained_model]['input_size']
+            if 'n_blocks' in self._models[pretrained_model]:
+                n_blocks = self._models[pretrained_model]['n_blocks']
+            if 'pyramids' in self._models[pretrained_model]:
+                pyramids = self._models[pretrained_model]['pyramids']
+            if 'mid_stride' in self._models[pretrained_model]:
+                mid_stride = self._models[pretrained_model]['mid_stride']
+            if 'mean' in self._models[pretrained_model]:
+                mean = self._models[pretrained_model]['mean']
+                self._use_pretrained_model = True
+
+        chainer.config.mid_stride = mid_stride
+        chainer.config.comm = comm
+
+        if initialW is None:
+            chainer.config.initialW = chainer.initializers.HeNormal()
+        else:
+            chainer.config.initialW = initialW
+
+        if not isinstance(input_size, (list, tuple)):
+            input_size = (int(input_size), int(input_size))
+
+        with self.init_scope():
+            self.input_size = input_size
+            self.trunk = DilatedFCN(n_blocks=n_blocks)
+
+            # To calculate auxirally loss
+            if chainer.config.train:
+                self.cbr_aux = ConvBNReLU(None, 512, 3, 1, 1)
```

Comment:
I do not like to change the configuration of a model using global variable.
It would not be much problem to instantiate these links. Why not always instantiate them?

Reply:
OK, let's instantiate them always.